test/integration/src/test/java/io/pravega/test/integration/BackUpRecoveryTest.java

```diff
@@ -0,0 +1,1220 @@
+
+        // Create a stream for writing data
+        createScopeStream(pravegaRunner.getControllerRunner().controller, SCOPE, STREAM1, config);
+
+        log.info("Created stream '{}'.", STREAM1);
+
+        // Create a client to write events.
+        try (val clientRunner = new ClientRunner(pravegaRunner.getControllerRunner())) {
+            // Write events.
+            writeEventsToStream(clientRunner.clientFactory, withTransaction);
+        }
+
+        log.info("Shutting down controller");
+        pravegaRunner.shutDownControllerRunner(); // Shut down the controller
+        // Flush DurableLog to Long Term Storage
+        log.info("Flush to storage");
+        flushToStorage(pravegaRunner.getSegmentStoreRunner().getServiceBuilder());
+
+        pravegaRunner.shutDownSegmentStoreRunner(); // Shutdown SegmentStore
+        pravegaRunner.shutDownBookKeeperRunner(); // Shutdown BookKeeper & ZooKeeper
+        pravegaRunner.close();
+        log.info("SegmentStore, BookKeeper & ZooKeeper shutdown");
+
+        Thread.sleep(1000);
+        // Start a new segment store and controller
+        PravegaRunner pravegaRunner1 = new PravegaRunner(bookieCount, containerCount);
+        pravegaRunner1.startBookKeeperRunner(instanceId++);
+        log.info("Started bk and zk. ");
+        val bkConfig1 = BookKeeperConfig.builder()
+                .with(BookKeeperConfig.ZK_ADDRESS, "localhost:" + pravegaRunner1.getBookKeeperRunner().getBkPort())
+                .with(BookKeeperConfig.BK_LEDGER_PATH, pravegaRunner1.getBookKeeperRunner().getLedgerPath())
+                .with(BookKeeperConfig.ZK_METADATA_PATH, pravegaRunner1.getBookKeeperRunner().getLogMetaNamespace())
+                .with(BookKeeperConfig.BK_ENSEMBLE_SIZE, 1)
+                .with(BookKeeperConfig.BK_WRITE_QUORUM_SIZE, 1)
+                .with(BookKeeperConfig.BK_ACK_QUORUM_SIZE, 1)
+                .build();
+        this.dataLogFactory = new BookKeeperLogFactory(bkConfig1, pravegaRunner1.getBookKeeperRunner().getZkClient().get(), this.executorService());
+        log.info("Starting segmentstore and controller ");
+        pravegaRunner1.startControllerAndSegmentStore(this.storageFactory, this.dataLogFactory);
+        log.info("Started segment store and controller again.");
+
+        log.info("Starting client to read Events ");
+        // Create the client with new controller.
+        try (val clientRunner = new ClientRunner(pravegaRunner1.getControllerRunner())) {
+            // Try reading all events again to verify that the recovery was successful.
+            readEventsFromStream(clientRunner.clientFactory, clientRunner.readerGroupManager);
+            log.info("Read all events again to verify that segments were recovered.");
+        }
+        FileUtils.deleteDirectory(new File(baseDir.getPath()));
+    }
+
+    /**
+     * Tests the data recovery scenario with transactional writer. Events are written using a transactional writer.
```

Comment:
are we using a transactional writer in this test?

Reply:
NA